### DISCUSSION TITLE
Should `graphql-env.d.ts` be `.gitignore`ed?

### DISCUSSION THREAD

**Discussion by bnussman** [0no-co/gql.tada#19] [Q&A] [answered]
This is a pretty simple question, and the title sums it up. Should the the output file (`graphql-env.d.ts` for example) be ignored by git? What do you recommend?

PS. Thank you for developing this tool! I'm working on implementing it in one of my projects right now! 🎉

**Comment by JoviDeCroock** [comment, marked as answer]
I would not `.gitignore` it, in case that someone doesn't have a working LSP/... and in CI this will make sure that everything typechecks correctly

**Reply by bnussman** [reply]
That makes a lot of sense. Thank you!

**Reply by wweaver** [reply]
How do you stop schema changes on a remote not causing everyone's local file being updated even though they aren't working on anything graphql related?

e.g. We are using storyblok. Something is changed there which updates the graphql schema. When someone is working on their code they all of a sudden now have a change to the graphql-env.d.ts unrelated to anything they are working on. Are they expected to commit that change even though it's not related to what they are doing?

We were contemplating adding it to the `.gitignore` to alleviate this. I'd expect all devs should have this setup working so I wouldn't have that concern.

**Reply by kitten** [reply]
@wwveaver I don't know enough about Storyblok to give a definitive answer to this, but it depends on whether the schema changes are coincidental or unstable.

If they're unstable, that's something we can potentially fix.

If they're coincidental then it's up to you. People are free to update it, if it's a committed file and it makes sense, or you can implement a CI workflow to generate it on the fly. Either should work just fine and we have no particular opinion or recommendations for this case
